packages/core/src/utils/rpc.ts

```diff
@@ -146,6 +146,20 @@ export const _eth_getTransactionReceipt = (
       return receipt as SyncTransactionReceipt;
     });
 
+/**
+ * Helper function for "eth_getBlockReceipts" request.
+ */
+export const _eth_getBlockReceipts = (
+  requestQueue: RequestQueue,
+  { blockHash }: { blockHash: Hash },
+): Promise<SyncTransactionReceipt[]> =>
+  requestQueue
+    .request({
+      method: "eth_getBlockReceipts",
```

Comment:
Just did some research on chain support.

- `eth_getBlockReceipts` is non-standard
- Erigon added it in April 2021 ([pr](https://github.com/erigontech/erigon/pull/1735)) and it started becoming popular
- Geth only added support in September 2023 ([release](https://github.com/ethereum/go-ethereum/releases/tag/v1.13.0))
- It's still somewhat [controversial](https://github.com/base-org/node/issues/125#issuecomment-1758344540)

TL;DR I strongly suspect we will encounter chains that don't support it, and we'll need to include a fallback that uses `eth_getTransactionReceipt`. Also may make sense to use that method anyway if the # of required transactions in the block is less than some threshold (2?).

Reply:
They are currently priced the same on alchemy

https://docs.alchemy.com/reference/compute-unit-costs#standard-evm-json-rpc-methods